From cf2d0a31414cc082720f0da39d4bc2b738f3a22b Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.tech>
Date: Thu, 28 Aug 2025 17:45:07 +0200
Subject: [PATCH] Fix tests of "CP-51843: add unit tests for setup_cache"

Upstream patch of ae103499e1db091e232051c7e4200ca7b39ce64c is incorrect.

All "@mock.patch('blktap2.VDI.PhyLink', autospec=True)" lines must be removed
because PhyLink is mocked globally.

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.tech>
---
 tests/test_blktap2.py | 28 +++++++---------------------
 1 file changed, 7 insertions(+), 21 deletions(-)

diff --git a/tests/test_blktap2.py b/tests/test_blktap2.py
index 0700dcd8..3d92355f 100644
--- a/tests/test_blktap2.py
+++ b/tests/test_blktap2.py
@@ -329,13 +329,11 @@ class TestVDI(unittest.TestCase):
     @mock.patch('blktap2.time.sleep', autospec=True)
     @mock.patch('blktap2.util.get_this_host', autospec=True)
     @mock.patch('blktap2.VDI._attach', autospec=True)
-    @mock.patch('blktap2.VDI.PhyLink', autospec=True)
     @mock.patch('blktap2.VDI.BackendLink', autospec=True)
     @mock.patch('blktap2.VDI.NBDLink', autospec=True)
     @mock.patch('blktap2.Tapdisk')
     def test_activate(self, mock_tapdisk, mock_nbd_link, mock_backend,
-                      mock_phy, mock_attach,
-                      mock_this_host, mock_sleep):
+                      mock_attach, mock_this_host, mock_sleep):
         """
         Test blktap2.VDI.activate, no cache, RW, success
         """
@@ -358,14 +356,12 @@ class TestVDI(unittest.TestCase):
     @mock.patch('blktap2.time.sleep', autospec=True)
     @mock.patch('blktap2.util.get_this_host', autospec=True)
     @mock.patch('blktap2.VDI._attach', autospec=True)
-    @mock.patch('blktap2.VDI.PhyLink', autospec=True)
     @mock.patch('blktap2.VDI.BackendLink', autospec=True)
     @mock.patch('blktap2.VDI.NBDLink', autospec=True)
     @mock.patch('blktap2.Tapdisk')
     def test_activate_relink_retry(
             self, mock_tapdisk, mock_nbd_link, mock_backend,
-            mock_phy, mock_attach,
-            mock_this_host, mock_sleep):
+            mock_attach, mock_this_host, mock_sleep):
         """
         Test blktap2.VDI.activate, relinking, retry 1, success
         """
@@ -386,14 +382,12 @@ class TestVDI(unittest.TestCase):
     @mock.patch('blktap2.time.sleep', autospec=True)
     @mock.patch('blktap2.util.get_this_host', autospec=True)
     @mock.patch('blktap2.VDI._attach', autospec=True)
-    @mock.patch('blktap2.VDI.PhyLink', autospec=True)
     @mock.patch('blktap2.VDI.BackendLink', autospec=True)
     @mock.patch('blktap2.VDI.NBDLink', autospec=True)
     @mock.patch('blktap2.Tapdisk')
     def test_activate_pause_retry(
             self, mock_tapdisk, mock_nbd_link, mock_backend,
-            mock_phy, mock_attach,
-            mock_this_host, mock_sleep):
+            mock_attach, mock_this_host, mock_sleep):
         """
         Test blktap2.VDI.activate, paused, retry 1, success
         """
@@ -413,14 +407,12 @@ class TestVDI(unittest.TestCase):
     @mock.patch('blktap2.time.sleep', autospec=True)
     @mock.patch('blktap2.util.get_this_host', autospec=True)
     @mock.patch('blktap2.VDI._attach', autospec=True)
-    @mock.patch('blktap2.VDI.PhyLink', autospec=True)
     @mock.patch('blktap2.VDI.BackendLink', autospec=True)
     @mock.patch('blktap2.VDI.NBDLink', autospec=True)
     @mock.patch('blktap2.Tapdisk')
     def test_activate_paused_while_tagging(
             self, mock_tapdisk, mock_nbd_link, mock_backend,
-            mock_phy, mock_attach,
-            mock_this_host, mock_sleep):
+            mock_attach, mock_this_host, mock_sleep):
         """
         Test blktap2.VDI.activate, paused, while tagging, success
         """
@@ -446,14 +438,12 @@ class TestVDI(unittest.TestCase):
     @mock.patch('blktap2.time.sleep', autospec=True)
     @mock.patch('blktap2.util.get_this_host', autospec=True)
     @mock.patch('blktap2.VDI._attach', autospec=True)
-    @mock.patch('blktap2.VDI.PhyLink', autospec=True)
     @mock.patch('blktap2.VDI.BackendLink', autospec=True)
     @mock.patch('blktap2.VDI.NBDLink', autospec=True)
     @mock.patch('blktap2.Tapdisk')
     def test_activate_relink_while_tagging(
             self, mock_tapdisk, mock_nbd_link, mock_backend,
-            mock_phy, mock_attach,
-            mock_this_host, mock_sleep):
+            mock_attach, mock_this_host, mock_sleep):
         """
         Test blktap2.VDI.activate, relinking, while tagging, retry 1, success
         """
@@ -479,14 +469,12 @@ class TestVDI(unittest.TestCase):
     @mock.patch('blktap2.time.sleep', autospec=True)
     @mock.patch('blktap2.util.get_this_host', autospec=True)
     @mock.patch('blktap2.VDI._attach', autospec=True)
-    @mock.patch('blktap2.VDI.PhyLink', autospec=True)
     @mock.patch('blktap2.VDI.BackendLink', autospec=True)
     @mock.patch('blktap2.VDI.NBDLink', autospec=True)
     @mock.patch('blktap2.Tapdisk')
     def test_activate_ro_already_activating_retry(
             self, mock_tapdisk, mock_nbd_link, mock_backend,
-            mock_phy, mock_attach,
-            mock_this_host, mock_sleep):
+            mock_attach, mock_this_host, mock_sleep):
         """
         If we're activating for read-only access, with someone else (let's
         say, another host in the pool) also being in the process of
@@ -517,14 +505,12 @@ class TestVDI(unittest.TestCase):
     @mock.patch('blktap2.time.sleep', autospec=True)
     @mock.patch('blktap2.util.get_this_host', autospec=True)
     @mock.patch('blktap2.VDI._attach', autospec=True)
-    @mock.patch('blktap2.VDI.PhyLink', autospec=True)
     @mock.patch('blktap2.VDI.BackendLink', autospec=True)
     @mock.patch('blktap2.VDI.NBDLink', autospec=True)
     @mock.patch('blktap2.Tapdisk')
     def test_activate_rw_already_activating_fail(
             self, mock_tapdisk, mock_nbd_link, mock_backend,
-            mock_phy, mock_attach,
-            mock_this_host, mock_sleep):
+            mock_attach, mock_this_host, mock_sleep):
         """
         If we're activating for read-write access, with someone else (let's
         say, another host in the pool) also being in the process of
